src/lib.rs

```diff
@@ -65,215 +98,2254 @@ pub struct Leser {
+#[xml(root = b"dokumentRepraesentationType")]
+pub struct DokumentRepraesentationType {
+    #[xml(ns = b"xwas", name = b"referenz", ty = "child")]
+    pub referenz: Option<String>,
+    #[xml(ns = b"xwas", name = b"mimeType", ty = "child")]
+    pub mime_type: String,
+    #[xml(ns = b"xwas", name = b"inhalt", ty = "child")]
+    pub inhalt: Option<String>,
+    #[xml(ns = b"xwas", name = b"externerReferenzTyp", ty = "child")]
+    pub externer_referenz_typ: Option<String>,
+    #[xml(ns = b"xwas", name = b"externerReferenzIndex", ty = "child")]
+    pub externer_referenz_index: Option<String>,
+    #[xml(ns = b"xwas", name = b"inhaltTyp", ty = "child")]
+    pub inhalt_typ: Option<String>,
+    #[xml(ns = b"xwas", name = b"signaturen", ty = "child")]
+    pub signaturen: Option<SignaturenType>,
+}
+
+/// Eine zum Antrag gehörige Unterlage in verschiedenen Dokumentendarstellungen, z. B.
+/// PDF oder eine Datendarstellung. Es muss immer das komplette Dokument mit allen
+/// Darstellungen übertragen werden.
+#[derive(Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[xml(root = b"DokumentType")]
+pub struct DokumentType {
+    #[xml(ns = b"xwas", name = b"dokumentTyp", ty = "child")]
+    pub dokument_typ: CodeDokumenttypType,
+    #[xml(ns = b"xwas", name = b"name", ty = "child")]
+    pub name: String,
+    #[xml(ns = b"xwas", name = b"aktuelleVersion", ty = "child")]
+    pub aktuelle_version: Option<String>,
+    #[xml(ns = b"xwas", name = b"letzteVersion", ty = "child")]
+    pub letzte_version: Option<String>,
+    #[xml(ns = b"xwas", name = b"dokumentRepraesentation", ty = "child")]
+    pub dokument_repraesentation: Vec<DokumentRepraesentationType>,
+    #[xml(ns = b"xwas", name = b"personReferenzID", ty = "child")]
+    pub person_referenz_id: Vec<String>,
+}
+
+#[derive(Default, Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[serde(tag = "t", content = "c")]
+pub enum VorgangType {
+    #[xml(ns = b"xwas", name = b"pruefbericht")]
+    Pruefbericht(PruefberichtType),
+    #[xml(ns = b"xwas", name = b"untersuchungsplan")]
+    Untersuchungsplan(UntersuchungsplanType),
+    #[xml(ns = b"xwas", name = b"olgBericht", ty = "child")]
+    OlgBericht(DokumentType),
+    #[default]
+    #[xml(ns = b"xwas", name = b"unknown")]
+    None,
+}
+
+extern crate raxb as _raxb;
+
 #[wasm_bindgen]
 pub fn create_quality_report_xml(data: QualityReport) -> Result<String, JsValue> {
-    Ok(raxb::ser::to_string_pretty_with_decl(&data).map_err(|err| JsValue::from_str(&err.to_string()))?)
+    Ok(raxb::ser::to_string_pretty_with_decl(&data)
+        .map_err(|err| JsValue::from_str(&err.to_string()))?)
 }
 
 #[wasm_bindgen]
 pub fn parse_quality_report_xml(xml: String) -> Result<QualityReport, JsValue> {
     Ok(raxb::de::from_str(&xml).map_err(|err| JsValue::from_str(&err.to_string()))?)
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_auftraggeber() {
```

Comment:
This test is not doing anything, should be removed as whole section. We need separate ticket on that.